### PR TITLE
WIP: Project:create --template and --initialize flags

### PR DESCRIPTION
**Pull request by shawnawsu**
Work in progress needs additional testing for edge cases and proper error handling. Initial work to achieve:

- Adding a --template arg to project:create that would pass a template URL to the options_url parameter of the subscription

- If no template is provided, using the /setup/catalog API on Accounts to fetch a catalog file and allow the user to pick a template from the CLI

- Adding a --initialize flag to project:create that would execute the initialization of your project for you from the project_options.initialize key, as Console does, if you supplied a template

This relies on updates to the [client php library](https://github.com/platformsh/platformsh-client-php).

### CODE REVIEW

File: src/Command/Project/ProjectCreateCommand.php [hunk 1]
```diff
@@ -78,8 +81,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
+        // Grab the url of the yaml file.
+        if (!empty($options['catalog_url'])) {
+            $options['catalog'] = $options['catalog_url'];
+        }
+        else if (!empty($options['catalog_url'])) {
```
Comment (bbujisic) [inline src/Command/Project/ProjectCreateCommand.php]: 'template_url'?

File: src/Command/Project/ProjectCreateCommand.php [hunk 2]
```diff
@@ -78,8 +81,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+            $options['catalog'] = $options['template_url'];
+        }
+        
         $subscription = $this->api()->getClient()
             ->createSubscription(
+                $options['catalog'],
```
Comment (bbujisic) [inline src/Command/Project/ProjectCreateCommand.php]: Push to the back of the arg list.

**Comment by shawnawsu** [comment]
We're moving forward with #831 so this can be closed.
